Dockerfile

```diff
@@ -1,24 +1,29 @@
-FROM node:14.18.0-alpine as build
+FROM node:16.16.0-alpine as build
 
 ARG COMMIT_SHA=<not-specified>
-ENV NODE_ENV=production
+ENV NPM_CONFIG_CACHE="/tmp"
 
 WORKDIR /build-dir
 
 COPY package.json .
 COPY package-lock.json .
 
-RUN npm ci
+# install also DEV dependencies to enable Typescript compilation
+RUN npm ci --production=false
 
 COPY . .
 
+# compile Typescript to Javascript
 RUN npm run build
 
+# remove files not needed in the final stage
+RUN rm -rf node_modules src tsconfig.tsbuildinfo tsconfig.json build.tsconfig.json
+
 RUN echo "mia_template_service_name_placeholder: $COMMIT_SHA" >> ./commit.sha
 
 ########################################################################################################################
 
-FROM node:14.18.0-alpine
+FROM node:16.16.0-alpine
```

Comment:
Using node 16.15, you should also add tini

Reply:
I added it.

I conducted my own research to understand the implications of adding `tini` and I agree with adding it.  
However, I was not able to find any resource explaining why it should be added for node 16.15+. Could you provide any article explaining the reasons underlying such decision?

Reply:
Search about tini on Google chat, there is a long thread where we spoke about it